R/kfold-helpers.R

```diff
@@ -75,27 +71,28 @@ kfold_split_random <- function(K = 10, N = NULL) {
 
 #' @rdname kfold-helpers
 #' @export
-kfold_split_balanced <- function(K = 10, x = NULL) {
+kfold_split_stratified <- function(K = 10, x = NULL) {
   stopifnot(
     !is.null(x),
-    length(unique(x)) == 2,
     K == as.integer(K),
     length(K) == 1,
     K > 1,
     K <= length(x)
   )
-  x <- as.integer(as.factor(x)) - 1
-  x0_id <- which(x == 0)
-  x1_id <- which(x == 1)
-  bins <- rep(NA, length(x))
```

Comment:
I think you still need  `bins <- rep(NA, length(x))` otherwise later you can't do `bins[xids] <- rep(1:K, ceiling(N/K), )[1:N]` because `bins` doesn't exist.